src/ts/njs_core.d.ts

```diff
@@ -35,11 +37,544 @@ type NjsByteString = string & {
+     *   Otherwise, the `Buffer` will be zero-filled.
+     * @param encoding The character encoding used for call to `buf.fill(fill, encoding)` while
+     *   initalizing. Defaults to`'utf8'`.
+     */
+    static alloc(size: number, fill?: NjsStringLike | Buffer | Uint8Array | number, encoding?: BufferEncoding): Buffer;
+    /**
+     * The same as `Buffer.alloc()`, with the difference that the memory allocated for the buffer
+     * is not initialized, the contents of the new buffer is unknown and may contain sensitive data.
+     *
+     * @param size The count of octets to allocate.
+     */
+    static allocUnsafe(size: number): Buffer;
+
+    /**
+     * Returns the byte length of the specified `value`, when encoded using `encoding`.
+     *
+     * @param value The value to test.
+     * @param encoding The character encoding used to evaluate `value` if `value` is a `string`.
+     *   Defaults to `'utf8'`.
+     */
+    static byteLength(value: NjsStringLike | Buffer | TypedArray | DataView | ArrayBuffer, encoding?: BufferEncoding): number;
+
+    /**
+     * Compares `buffer1` with `buffer2` when sorting arrays of Buffer instances.
+     *
+     * @return
+     * - `0` if `buffer2` is the same as `buffer1`,
+     * - `1` if `buffer2` should come _before_ `buffer1` when sorted,
+     * - `-1` if `buffer2` should come _after_ `buffer1` when sorted.
+     */
+    static compare(buf1: Buffer, buf2: Buffer): -1 | 0 | 1;
+
+    /**
+     * Returns a new `Buffer` which is the result of concatenating all the `Buffer` instances in
+     * the `list`. If there are no items in the `list` or the total length is 0, a new zero-length
+     * `Buffer` is returned.
+     *
+     * @param list An array of `Buffer` objects to concatenate.
+     * @param totalLength Total length of the buffers when concatenated, coerced to an unsigned
+     *   integer. If not specified, it is calculated from the `Buffer` instances in `list` by adding
+     *   their lengths. If the combined length of the Buffers in list exceeds `totalLength`, the
+     *   result is truncated to `totalLength`.
+     */
+    static concat(list: Buffer[], totalLength?: number): Buffer;
+
+    /**
+     * Allocates a new `Buffer` using an array of bytes in the range `0 – 255`. Array entries
+     * outside that range will be truncated.
+     *
+     * @param data The data to create a new `Buffer`.
+     */
+    static from(data: Uint8Array | number[]): Buffer;
+    /**
+     * Copies the passed buffer `data` onto a new `Buffer` instance.
+     *
+     * @param data The buffer to copy.
+     */
+    static from(data: Buffer): Buffer;
+    /**
+     * For objects whose `valueOf()` function returns a value not strictly equal to object, returns
+     * `Buffer.from(object.valueOf(), offsetOrEncoding, length)`.
+     *
+     * @param obj An object supporting `Symbol.toPrimitive` or `valueOf()`.
```

Comment:
`Symbol.toPrimitive` is not supported.

Reply:
Actually, it seems that not even `valueOf` is implemented.

```
>> Buffer.from(new Date())              
Thrown:
TypeError: first argument date is not a string or Buffer-like object
    at Buffer.from (native)
    at main (shell:1)
```

or 

```
>> Buffer.from({ valueOf: () => 'foo' })
Thrown:
TypeError: first argument object is not a string or Buffer-like object
    at Buffer.from (native)
    at main (shell:1)
```

However, there is the code path for an object…

https://github.com/nginx/njs/blob/976ed440e983f56a96630caae9103f1cf80345cf/src/njs_buffer.c#L252-L258

But if I understand `njs_buffer_from_object` correctly, it accepts only one argument instead of 3 (`Buffer.from(object[, offsetOrEncoding[, length]])`):

https://github.com/nginx/njs/blob/976ed440e983f56a96630caae9103f1cf80345cf/src/njs_buffer.c#L267-L370

Reply:
yes, this was missed. @lexborisov  is on it.